addons/website/static/src/scss/website.scss

```diff
@@ -1516,6 +1516,28 @@ span.list-inline-item.o_add_language:last-child {
     @include text-emphasis-variant(".text-#{$media}", $color);
 }
 
+// Modal
+.modal:not(.o_technical_modal) .modal-content {
+    // If the text color of the body (used for the text color in modals) is not
+    // visible due to insufficient contrast with the modal background, we adjust
+    // the text color in the modal using the following code. For example, if the
+    // user sets a black background for its website and the text color of the
+    // body is white, the text will not be visible on modals with a white
+    // background.
+    @if $modal-content-color == null {
+        color: if(
+            color-yiq($body-color) == $yiq-text-dark,
+            color-yiq($modal-content-bg, $light: $body-color, $dark: $yiq-text-dark),
+            color-yiq($modal-content-bg, $light: $yiq-text-light, $dark: $body-color)
+        );
+    }
+    @if color-yiq($body-color) == $yiq-text-dark {
+        .text-muted {
+            color: color-yiq($modal-content-bg, $light: $text-muted, $dark: mute-color($yiq-text-dark)) !important;
```

Comment:
Not handling situations were `user_custom_bootstrap_overridden` contains an override of `$modal-content-bg` to make modals dark too, because this then becomes a matter of customizing all involved variables in a consistent way.